04_testing/outlier_checks.R

```diff
@@ -897,3 +898,365 @@ write_csv(
+    x = "Population size, disaggregated area",
+    y = "% of areas",
+    title = "% of units of analysis by population size"
+  ) +
+  theme_minimal() +
+  theme(
+    plot.title = element_text(
+      face = "bold",
+      size = 22,
+      margin = margin(10, 10, 10, 10, "pt"),
+      family = "Roboto"
+    ),
+    plot.background = element_rect(
+      fill = "white"
+    ),
+    axis.text = element_text(
+      face = "bold",
+      size = 10,
+      family = "Roboto"
+    ),
+    legend.text = element_text(
+      size = 12,
+      family = "Roboto"
+    ),
+    legend.position = "bottom",
+    panel.grid.minor = element_blank(),
+    legend.background = element_rect(fill = "transparent"),
+    legend.box.background = element_rect(fill = "transparent"),
+    strip.text = element_text(
+      size = 16,
+      family = "Roboto"
+    )
+  )
+
+ggsave(
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "2022_population_histogram.png"
+  ),
+  height = 6,
+  width = 8
+)
+
+write_csv(
+  df_pop_hist,
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "datasets",
+    "2022_population_histogram.csv"
+  )
+)
+
+## Look at number of units of analysis and outliers
+
+df_outliers_num <- df_outliers %>%
+  group_by(
+    adm0_pcode
+  ) %>%
+  summarize(
+    units = n(),
+    num_sectors = length(unique(sector)),
+    pct_outliers = sum(is_any_outlier) / units,
+    .groups = "drop"
+  )
+
+df_outliers_num %>%
+  ggplot() +
+  geom_point(
+    aes(
+      x = units,
+      y = pct_outliers
+    ),
+    size = 3,
+    color = "#1EBFB3"
+  ) +
+  scale_x_log10() +
+  scale_y_continuous(
+    labels = scales::percent_format(1)
+  ) +
+  theme_minimal() +
+  theme(
+    plot.title = element_text(
+      face = "bold",
+      size = 22,
+      margin = margin(10, 10, 10, 10, "pt"),
+      family = "Roboto"
+    ),
+    plot.background = element_rect(
+      fill = "white"
+    ),
+    axis.text = element_text(
+      face = "bold",
+      size = 10,
+      family = "Roboto"
+    ),
+    legend.text = element_text(
+      size = 12,
+      family = "Roboto"
+    ),
+    legend.position = "bottom",
+    panel.grid.minor = element_blank(),
+    legend.background = element_rect(fill = "transparent"),
+    legend.box.background = element_rect(fill = "transparent"),
+    strip.text = element_text(
+      size = 16,
+      family = "Roboto"
+    )
+  ) +
+  labs(
+    x = "# of units of analysis (log10 scale)",
+    y = "% of outliers",
+    title = "Outliers by number of units of analysis"
+  )
+
+ggsave(
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "2022_outliers_number_units_analysis.png"
+  ),
+  height = 4,
+  width = 7
+)
+
+write_csv(
+  df_outliers_num,
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "datasets",
+    "2022_outliers_number_units_analysis.csv"
+  )
+)
+
+# also plotting against # of sectors
+
+df_outliers_num %>%
```

Comment:
same here, showing the countries as legend will explain the other outlier graphs.

Reply:
maybe adding a note, normally for outlier detection, the more cases available the more accurate outlier detection will be. I know that it but don't have any proof for it yet 😆 , if you think it's correct having that will also validates the crazy numbers for countries with only the top common sectors.

Reply:
Hmm from the graph, it appears that there's very few outliers if we only have a small # of sectors. What do we think about that?